hera_mc/geo_handling.py

```diff
@@ -127,23 +127,27 @@ def get_station_types(self, add_stations=True):
         add_stations:  if True, add all of the stations to their types
                        if False, just return station types
         """
-
+        if self.__stations_added_to_types:
+            return
+        if not add_stations and self.station_types is not None:
+            return
         station_data = self.session.query(geo_location.StationType).all()
         stations = {}
         for sta in station_data:
```

Comment:
This isn't a change in this PR, but it is better to write the loop as:

```python
for sta in self.session.query(geo_location.StationType):
```

While the `.all()` call buffers the results of the database query into one big list, iterating over the non-`.all()` query streams through the database results as they come back, which is more efficient. Not that the difference is large in this particular case. And in my opinion, avoiding the extra local variable here makes things somewhat clearer.

Reply:
Did this, and then the very next loop was same, so changed too.